boa/src/syntax/ast/node.rs

```diff
@@ -0,0 +1,315 @@
+                if let Some(func_name) = name {
+                    f.write_fmt(format_args!("{}", func_name))?;
+                }
+                write!(f, "{{")?;
+                // join_expr(f, args)?;
+                write!(f, "}} {}", expr)
+            }
+            Node::ArrowFunctionDecl(ref _args, ref expr) => {
+                write!(f, "(")?;
+                // join_expr(f, args.ini)?;
+                write!(f, ") => {}", expr)
+            }
+            Node::BinOp(ref op, ref a, ref b) => write!(f, "{} {} {}", a, op, b),
+            Node::UnaryOp(ref op, ref a) => write!(f, "{}{}", op, a),
+            Node::Return(Some(ref ex)) => write!(f, "return {}", ex),
+            Node::Return(None) => write!(f, "return"),
+            Node::Throw(ref ex) => write!(f, "throw {}", ex),
+            Node::Assign(ref ref_e, ref val) => write!(f, "{} = {}", ref_e, val),
+            Node::VarDecl(ref vars) | Node::LetDecl(ref vars) => {
+                if let Node::VarDecl(_) = *self {
+                    f.write_str("var ")?;
+                } else {
+                    f.write_str("let ")?;
+                }
+                for (key, val) in vars.iter() {
+                    match val {
+                        Some(x) => f.write_fmt(format_args!("{} = {}", key, x))?,
+                        None => f.write_fmt(format_args!("{}", key))?,
+                    }
+                }
+                Ok(())
+            }
+            Node::ConstDecl(ref vars) => {
+                f.write_str("const ")?;
+                for (key, val) in vars.iter() {
+                    f.write_fmt(format_args!("{} = {}", key, val))?
+                }
+                Ok(())
+            }
+            Node::TypeOf(ref e) => write!(f, "typeof {}", e),
+        }
+    }
+}
+
+/// `join_expr` - Utility to join multiple Expressions into a single string
+fn join_expr(f: &mut Formatter, expr: &[Node]) -> Result {
+    let mut first = true;
+    for e in expr.iter() {
+        if !first {
+            f.write_str(", ")?;
+        }
+        first = false;
+        Display::fmt(e, f)?;
+    }
+    Ok(())
+}
+
+/// "Formal parameter" is a fancy way of saying "function parameter".
+///
+/// In the declaration of a function, the parameters must be identifiers,
+/// not any value like numbers, strings, or objects.
+///```javascript
+///function foo(formalParametar1, formalParametar2) {
+///}
+///```
+/// For more information, please check <https://tc39.es/ecma262/#prod-FormalParameter>
+#[derive(Clone, Debug, PartialEq, Trace, Finalize)]
+pub struct FormalParameter {
+    pub name: String,
+    pub init: Option<Box<Node>>,
+    pub is_rest_param: bool,
+}
+
+pub type FormalParameters = Vec<FormalParameter>;
+
+impl FormalParameter {
+    pub fn new(name: String, init: Option<Box<Node>>, is_rest_param: bool) -> FormalParameter {
+        FormalParameter {
+            name: name,
+            init: init,
+            is_rest_param: is_rest_param,
```

Comment:
This solves the clippy error.

```suggestion
            name,
            init,
            is_rest_param,
```